components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/AdminRoleListener.java

```diff
@@ -124,7 +125,12 @@ public void postGetPermissionListOfRoles(List<String> permissions, List<String>
             try {
                 List<Scope> systemScopes = ApplicationManagementServiceComponentHolder.getInstance()
                         .getAPIResourceManager().getSystemAPIScopes(tenantDomain);
-                permissions.addAll(systemScopes.stream().map(Scope::getName).collect(Collectors.toList()));
+                permissions.addAll(
+                        systemScopes.stream()
+                                .map(Scope::getName)
+                                // Impersonate scope should not be given by default even for an admin.
+                                .filter(name -> !IMPERSONATE_SCOPE_NAME.equals(name))
```

Comment:
Can we have this as a configurable list? Scope that shouldn't be assigned by default.

Reply:
+1

Reply:
Will add these changes in a different PR
https://github.com/wso2/carbon-identity-framework/pull/6690
Hence removing the commit.